src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -341,72 +343,97 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                             relativePath = eventPaths[i].Remove(0, _fullDirectory.Length);
                         }
 
-                        // Check if this is a rename
-                        if (IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
+                        var changedFlags = FSEventStreamEventFlags.kFSEventStreamEventFlagItemInodeMetaMod |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemFinderInfoMod |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemModified |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemChangeOwner |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemXattrMod;
+
+                        // If any of the Changed flags are set in both Filter and Event, return true as a Changed event has occurred.
```

Comment:
This is a stale comment. I used to have these checks mirrored in an ```EventPassesFilters``` helper method at the beginning of the "else if" but removed that method because there was too much duplication.